### ISSUE TITLE
Bug: SSH is not working over OpenSsl 3.0

### ISSUE THREAD

**Issue by 134130** [libgit2/libgit2#6426]
First of all, I would like to thank everyone who maintains this project.

I'm using libgit2 with libgit2sharp.
Since i'm using mac, I build native libraries with docker buildx.

### Reproduction steps
Build with dockerx build
```
FROM ubuntu

ARG DEBIAN_FRONTEND=noninteractive

RUN apt-get update
RUN apt-get install -y -f --no-install-recommends \
    cmake gcc make ninja-build libz-dev \
    libkrb5-dev libssl-dev libssh2-1-dev libcurl4-openssl-dev \
    pkgconf python2 python3

COPY native /root/native

RUN mkdir /root/native/libgit2/runtimes
WORKDIR /root/native/libgit2/runtimes

RUN cmake -DCMAKE_BUILD_TYPE:STRING=Release \
                         -DLIBGIT2_FILENAME=git2 \
                         -DUSE_SSH=ON \
                         -DUSE_HTTPS=ON \
                         -DUSE_BUNDLED_ZLIB=ON \
                         ..

RUN cmake --build .
```

WIth this script, 
```
#12 44.72 -- Found OpenSSL: /usr/lib/aarch64-linux-gnu/libcrypto.so (found version "3.0.2") 
```
```
#13 664.5 /root/native/libgit2/deps/ntlmclient/crypt_openssl.c: In function 'ntlm_crypt_init_functions':
#13 664.5 /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:128:9: warning: 'DES_set_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
#13 664.5   128 |         ntlm->crypt_ctx.des_set_key_fn = DES_set_key;
#13 664.5       |         ^~~~
#13 664.5 In file included from /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:16:
#13 664.5 /usr/include/openssl/des.h:186:5: note: declared here
#13 664.5   186 | int DES_set_key(const_DES_cblock *key, DES_key_schedule *schedule);
#13 664.5       |     ^~~~~~~~~~~
#13 664.5 /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:129:9: warning: 'DES_ecb_encrypt' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
#13 664.5   129 |         ntlm->crypt_ctx.des_ecb_encrypt_fn = DES_ecb_encrypt;
#13 664.5       |         ^~~~
#13 664.5 In file included from /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:16:
#13 664.5 /usr/include/openssl/des.h:105:6: note: declared here
#13 664.5   105 | void DES_ecb_encrypt(const_DES_cblock *input, DES_cblock *output,
#13 664.5       |      ^~~~~~~~~~~~~~~
#13 664.5 /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:133:9: warning: 'HMAC_CTX_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
#13 664.5   133 |         ntlm->crypt_ctx.hmac_ctx_new_fn = HMAC_CTX_new;
#13 664.5       |         ^~~~
#13 664.5 In file included from /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:18:
#13 664.5 /usr/include/openssl/hmac.h:33:33: note: declared here
#13 664.5    33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
#13 664.5       |                                 ^~~~~~~~~~~~
#13 664.5 /root/native/libgit2/deps/ntlmclient/crypt_openssl.c:134:9: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
#13 664.5   134 |         ntlm->crypt_ctx.hmac_ctx_free_fn = HMAC_CTX_free;
#13 664.5       |         ^~~~
```

These warnings are presented when building. (so many)

When using `git_remote_connect()` with ssh credentials, It occurs an error

On MacOS, I uses LibreSSL 2.8.3. So there's no errors

### Expected behavior
Works well

### Actual behavior
Error: this remote has never connected

### Version of libgit2 (release number or SHA1)
1.4.4
There is no meaningful diff on 1.4.4 and 1.5.0. on crypt_openssl.c

### Operating system(s) tested
Amazon linux

**Comment by carlosmn** [comment]
Those warnings you paste are all about a different mechanism. OpenSSL 3 has deprecated a whole bunch of things so warnings about that is expected.

You did not specify what error you saw, so we can't know if the error is even something that libgit2 has any control over.

This doesn't seem to be an issue with libgit2 though, but libssh2 not working right with OpenSSL 3. At work we've experienced issues with libssh and OpenSSL 3 so it's quite possible that libssh2 has similar issues.

**Comment by ethomson** [comment]
The warnings that you see are fixed in #6464 and #6471. If you're still having trouble with libssh2, please open an issue with those details. Thanks!
